app/lib/tufts/workflow_status.rb

```diff
@@ -0,0 +1,33 @@
+module Tufts
+  module WorkflowStatus
+    def self.status(id)
+      work_id = id
+      solr_doc_status = solr_status(work_id)
+      if solr_doc_status == 'unpublished' || solr_doc_status == 'deposited'
+        'unpublished'
+      else
+        # Only look for this status in Fedora if necessary
+        work = ActiveFedora::Base.find(id)
```

Comment:
We shouldn't need to hit Fedora, since it doesn't have any data we are interested in. `ActiveFedora::Base.new(id: id).draft_saved?` should get the correct result without reading anything except `File.exist?`